File: 卡码/代码随想录/JS/动态规划/01-基础题目/LeetCode_0343.js
//343. 整数拆分
const integerBreak = function(n) {
	const dp=new Array(n+1).fill(0);
	dp[0]=0,dp[1]=0,dp[2]=1;
	for(let i=3;i<n+1;i++){
		for(let j=1;j<=Math.floor(i/2);j++){
			dp[i]=Math.max(dp[i],Math.max(j*(i-j),j*dp[i-j]));
		}
	}
	return dp[n];
};

File: 卡码/代码随想录/Java/codeThoughts/src/main/java/com/kurumi/leetCode/codeThoughts/动态规划/_01_基础题目/LeetCode_0343.java
package com.kurumi.leetCode.codeThoughts.动态规划._01_基础题目;

/**
 * @Author kurumi
 * @Date 2026-07-03 上午11:52:20
 * @Description 343. 整数拆分
 */
public class LeetCode_0343 {
    static class Solution {
        public int integerBreak(int n) {
            int[] dp = new int[n + 1];
            dp[0] = 0;
            dp[1] = 0;
            dp[2] = 1;
            for (int i = 3; i <= n; i++)
                for (int j = 1; j <= i / 2; j++) {
                    dp[i] = Integer.max(dp[i], Integer.max(j * dp[i - j], j * (i - j)));
                }
            return dp[n];
        }
    }
}
